modules/calib3d/src/calibinit.cpp

```diff
@@ -451,11 +451,16 @@ int cvFindChessboardCorners( const void* arr, CvSize pattern_size,
     if( !out_corners )
         CV_Error( CV_StsNullPtr, "Null pointer to corners" );
 
-    if (img.channels() != 1)
+    if (img.channels() == 3)
     {
         cvtColor(img, img, COLOR_BGR2GRAY);
```

Comment:
COLOR_RGB2GRAY and COLOR_BGR2GRAY are not equivalent, so changing them doesn't work.
BGR is the current color format for imread/imshow functions (on Windows/Linux/Mac).
RGBA is used on Android for camera capture, so be careful. There is some issue with "description" about this.